File: aoc2019/day1/part1.ts
import fs from "fs";

const main = () => {
  try {
    const data = fs.readFileSync("./input.txt", "utf-8");
    const arr = data
      .split("\n")
      .map((item) => parseInt(item))
      .filter(Boolean);
    let fuelRequired = 0;
    arr.forEach((item) => {
      item = Math.floor(parseFloat(item) / 3);
      item = item - 2;
      fuelRequired += item;
    });
    console.log(fuelRequired);
  } catch (err) {
    console.error(err);
  }
};

main();

File: aoc2019/day1/part1.py
import math


def main():
    with open('input.txt', 'r') as f:
        p = [int(i.strip()) for i in f]
    # print(p)
    fuelRequired = 0
    for item in p:
        item = math.floor(float(item)//3)
        item = item - 2
        fuelRequired += item
    print(fuelRequired)


main()
